types/accept/index.d.ts

```diff
@@ -0,0 +1,21 @@
+// Type definitions for accept 3.1
+// Project: https://github.com/hapijs/accept#readme
+// Definitions by: feinoujc <https://github.com/feinoujc>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+export function charset(charsetHeader?: string, preferences?: string[]): string;
```

Comment:
Are the headers really optional?  The [API Reference](https://github.com/hapijs/accept/blob/master/API.md) makes them seem mandatory.

Reply:
I think they should be optional. a common usage would be 

```ts
accept.language(headers['accept-language']);
```

this is handled for example [here](https://github.com/hapijs/accept/blob/9046711d2bc91f0fc2a68aa1b767322b7a5d7b97/lib/language.js#L54)

Reply:
I think I'm missing something - isn't your example usage passing the header parameter?

Superficially, the handling code you link appears to be hardening against bad imports.  Since TS has compile-time checking, it seems like preventing bad the call from being made in the first place would be preferable.

Reply:
the headers object here would likely be some derivative of node's `http.IncomingHttpHeaders` where everything is optional, so if I make it not optional it would force some unnecessary checks or casting that is handled by the internals of the library anyway.

```ts
interface IncomingHttpHeaders {
        'accept'?: string;
        'accept-patch'?: string;
      // etc...
}```